### ISSUE TITLE
Update screen-reader-text class CSS

### ISSUE THREAD

**Issue by samikeijonen** [Automattic/_s#1211]
[WP Accessibility repo](https://github.com/wpaccessibility/a11ythemepatterns/blob/master/read-more-links/style.css) already have updated CSS for `screen-reader-text`. There is also trac ticket [40970](https://core.trac.wordpress.org/ticket/40970) for the same update. 

In short we are adding `clip-path` for newer browsers but still leaving the `clip` (which is deprecated) for older browsers. Also adding `margin: -1px` to hide it completely from sight.

Suggested CSS:

```css
.screen-reader-text {
	border: 0;
	clip: rect(1px, 1px, 1px, 1px);
	clip-path: inset(50%);
	height: 1px;
	margin: -1px;
	overflow: hidden;
	padding: 0;
	position: absolute !important;
	width: 1px;
	word-wrap: normal !important; /* Many screen reader and browser combinations announce broken words as they would appear visually. */
}
``` 

Suggested focus styles:

```css
.screen-reader-text:focus {
	background-color: #f1f1f1;
	border-radius: 3px;
	box-shadow: 0 0 2px 2px rgba(0, 0, 0, 0.6);
	clip: auto !important;
        clip-path: none;
	color: #21759b;
	display: block;
	font-size: 14px;
	font-size: 0.875rem;
	font-weight: bold;
	height: auto;
	left: 5px;
	line-height: normal;
	padding: 15px 23px 14px;
	text-decoration: none;
	top: 5px;
	width: auto;
	z-index: 100000; /* Above WP toolbar. */
}
```

**Comment by m-e-h** [comment]
@samikeijonen Have you seen the bug and ongoing discussion with h5bp's similar css?  https://github.com/h5bp/html5-boilerplate/issues/1985

**Comment by samikeijonen** [comment]
Thanks for pointing that out! More testing coming up.

**Comment by grappler** [comment]
`npm install stylelint-config-wordpress`

**Comment by samikeijonen** [comment]
There still seem to be a bug in Voiceover which announce words in wrong order. But in [Core there is already new `screen-reader-text` class coming in 4.9](https://make.wordpress.org/core/2017/10/22/changes-to-the-screen-reader-text-css-class-in-wordpress-4-9/).

PR #1218 use similar method and is the same as I proposed in this issue originally.

**Comment by grappler** [comment]
This is has been [reported to Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=757177) but as this issue is also in Safari I think this is more of a VoiceOver bug which needs to be fixed by Apple but I am not sure how that would work.

**Comment by samikeijonen** [comment]
Yeap it needs to be fixed by Apple. Note that this PR doesn't change the order bug. It was there already before this PR.

**Comment by davidakennedy** [comment]
Seems like, despite the browser issues, this should be okay to merge.

**Comment by davidakennedy** [comment]
Fixed in #1218.
